src/QsCompiler/SyntaxProcessor/ExpressionVerification.fs

```diff
@@ -362,7 +362,7 @@ let rec internal verifyBinding (inference: InferenceContext) tryBuildDeclaration
     | Symbol name ->
         match tryBuildDeclaration (name, symbol.RangeOrDefault) rhsType with
         | Some declaration, diagnostics -> VariableName name, [| declaration |], diagnostics
-        | None, diagnostics -> VariableName name, [||], diagnostics
+        | None, diagnostics -> InvalidItem, [||], diagnostics
```

Comment:
Just double-checking that this line is correct. We want `InvalidItem` here instead of `VariableName name`?

Reply:
None here means that the variable could not be added to the symbol tracker; i.e. the variable already exists and its redeclaration is invalid. In that case we generate a diagnostic. Not replacing the AST node here (keeping it as VariableName) makes the compiler throw when you e.g. hover over invalid code (like e.g. `(a, a) -> a`). This was an oversight before, and I added a unit test.